pkg/provisioner/ironic/ironic.go

```diff
@@ -277,24 +277,38 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 			}
 		}
 
-		if p.host.Spec.Image != nil && p.host.Spec.Image.URL != "" {
-			// FIXME(dhellmann): The Stein version of Ironic supports passing
-			// a URL. When we upgrade, we can stop doing this work ourself.
-			checksum, err := p.getImageChecksum()
-			if err != nil {
-				return result, errors.Wrap(err, "failed to retrieve image checksum")
-			}
+		// If there is an image to be provisioned, or an image has
+		// previously been provisioned, include those details. Either
+		// case may mean we are re-adopting a host that was already
+		// known but removed/lost because the pod restarted.
+		//
+		// FIXME(dhellmann): The Stein version of Ironic supports
+		// passing a URL for the checksum. When we upgrade, we can
```

Comment:
We already have Stein, trying to move to Train now.

Reply:
OK, good. I'll track that change separately.

Reply:
See #298